File: Java/DesignPatterns/src/test/java/designPatterns/behavioral/command/MainCommandTest.java
package designPatterns.behavioral.command;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MainCommandTest {
  private Receiver receiver;
  private Command commandA;
  private Command commandB;
  private Invoker invoker;

  @BeforeEach
  void setUp() {
    receiver = new Receiver();
    commandA = new ConcreteCommand.A(receiver);
    commandB = new ConcreteCommand.B(receiver);
    invoker = new Invoker();
  }

  @Test
  void equalsTest() {
    invoker.setCommand(commandA);
    assertThat(invoker.invokeExecute()).isEqualTo("A");

    invoker.setCommand(commandB);
    assertThat(invoker.invokeExecute()).isEqualTo("B");
  }
}

File: Kotlin/DesignPatterns/src/test/kotlin/behavioral/command/MainCommandTest.kt
package behavioral.command

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class MainCommandTest {
  private lateinit var receiver: Receiver
  private lateinit var commandA: Command
  private lateinit var commandB: Command
  private lateinit var invoker: Invoker

  @BeforeEach
  internal fun setUp() {
    receiver = Receiver()
    commandA = ConcreteCommand.A(receiver)
    commandB = ConcreteCommand.B(receiver)
    invoker = Invoker()
  }

  @Test
  internal fun equalsTest() {
    invoker.command = commandA
    assertThat(invoker.invokeExecute()).isEqualTo("A")

    invoker.command = commandB
    assertThat(invoker.invokeExecute()).isEqualTo("B")
  }
}